* compat hacks in case libc and kernel get out of sync:
* 
* glibc 2.4 and uClibc 0.9.29 introduce IPV6_RECVPKTINFO etc. and
*  change IPV6_PKTINFO enumeration.
* This is only supported in Linux kernel >= 2.6.14 
*
diff -urB uClibc-0.9.29/libc/sysdeps/linux/common/bits/in.h uClibc/libc/sysdeps/linux/common/bits/in.h
--- uClibc-0.9.29/libc/sysdeps/linux/common/bits/in.h	2006-02-22 11:48:45.000000000 +0300
+++ uClibc/libc/sysdeps/linux/common/bits/in.h	2009-04-05 15:05:52.000000000 +0400
@@ -130,6 +130,9 @@
 #define IPV6_V6ONLY		26
 #define IPV6_JOIN_ANYCAST	27
 #define IPV6_LEAVE_ANYCAST	28
+
+ #include <linux/version.h>
+ #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,14)
 #define IPV6_IPSEC_POLICY	34
 #define IPV6_XFRM_POLICY	35
 
@@ -147,6 +150,14 @@
 
 #define IPV6_RECVTCLASS		66
 #define IPV6_TCLASS		67
+ #else	/* 2.4.x kernels */
+#define IPV6_PKTINFO		IPV6_2292PKTINFO
+#define IPV6_HOPOPTS		IPV6_2292HOPOPTS
+#define IPV6_DSTOPTS		IPV6_2292DSTOPTS
+#define IPV6_RTHDR		IPV6_2292RTHDR
+#define IPV6_PKTOPTIONS		IPV6_2292PKTOPTIONS
+#define IPV6_HOPLIMIT		IPV6_2292HOPLIMIT
+ #endif
 
 /* Obsolete synonyms for the above.  */
 #define IPV6_ADD_MEMBERSHIP	IPV6_JOIN_GROUP
